### ISSUE TITLE
Cann't reproduce the result.

### ISSUE THREAD

**Issue by LtECoD** [zhaochaocs/DualEnc#1]
Hi,
    I have some problems in reproducing results in the paper. The reproduce process is as the fllowing.

Since no preprocess commands exists in script "pipeline_PlanEnc.sh" to read raw webnlg data. I find and run the commands in 'planning.sh': 
```bash
cd data/webnlg/
python3 ../../webnlg_eval_scripts/webnlg_gcnonmt_input.py -i ./ -c all -l -e
python3 ../../webnlg_eval_scripts/webnlg_gcnonmt_input.py -i ./ -p test -c all -l -e
cd ../..
```
Next, I follow the instruction to train & test the PlanEnc model 
`sh pipeline_PlanEnc.sh`, but it requires a little corrections. The corrected script is :
```bash
seed=1
echo "begin to train ... Loop :: " $seed

python3 train.py -data data/${model_id}_exp -save_model data/${save_model_name} \
            -encoder_type rnn -rnn_type LSTM  -layers 2 \
            -word_vec_size ${emb_size} -rnn_size ${hidden_size} \
            -epochs 20 -optim adam -learning_rate 0.001 -learning_rate_decay 0.7 -seed $seed\
            -gpuid 0 -start_checkpoint_at 15 \
            -copy_attn -brnn

echo "Begin to translate..."

model_path=$(head -1 data/${save_model_name}.txt)

python3 transform_plan.py --id_file data/webnlg/test-webnlg-all-${if_delex}-translate.lexid.txt \
        --plan_txt_file data/webnlg/test-webnlg-all-${if_delex}.triple \
        --plan_order_file data/webnlg/test-webnlg-all-${if_delex}-src-rel-order.txt \
        --new_plan_txt_file data/webnlg/test-webnlg-all-${if_delex}-plan.txt \
        -e -r -se -sr

# translating
python3 translate.py -model ${model_path} -data_type ${data_type} \
            -src data/webnlg/${eval_data}-webnlg-all-${if_delex}-plan.txt \
            -src2 data/webnlg/${eval_data}-webnlg-all-${if_delex}-plan.txt \
            -src_triple data/webnlg/${eval_data}-webnlg-all-${if_delex}.triple \
            -output data/webnlg/pred-${eval_data}-webnlg-all-${encoder}-${if_delex}.txt \
            -gpu 0 -batch_size 256 \
            -replace_unk -report_bleu


cd data/webnlg/
python3 ../../webnlg_eval_scripts/webnlg_gcnonmt_relexicalise.py -i ./ -p ${eval_data} -c all \
    -f pred-${eval_data}-webnlg-all-${encoder}-${if_delex}.txt -l
cd ../..
```
as you see, I fix the random seed (which is not crucial in my thoughts) to 1.

And finally, I run the script 'cauculate_bleu_test.sh' in webnlg_eval_scripts folder:
```bash
# compute BLEU
encoder=rnn
# if delex: delex; notdelex
if_delex=notdelex
# eval data: dev; test
eval_data=test
# part: seen; unseen; all
part=all

export TEST_TARGETS_REF0=data/webnlg/${eval_data}-${part}-notdelex-reference0.lex
export TEST_TARGETS_REF1=data/webnlg/${eval_data}-${part}-notdelex-reference1.lex
export TEST_TARGETS_REF2=data/webnlg/${eval_data}-${part}-notdelex-reference2.lex
export TEST_TARGETS_REF3=data/webnlg/${eval_data}-${part}-notdelex-reference3.lex
export TEST_TARGETS_REF4=data/webnlg/${eval_data}-${part}-notdelex-reference4.lex
export TEST_TARGETS_REF5=data/webnlg/${eval_data}-${part}-notdelex-reference5.lex
export TEST_TARGETS_REF6=data/webnlg/${eval_data}-${part}-notdelex-reference6.lex
export TEST_TARGETS_REF7=data/webnlg/${eval_data}-${part}-notdelex-reference7.lex

export TEST_PRED=data/webnlg/pred-${eval_data}-webnlg-${part}-${encoder}-${if_delex}-relex.txt

./webnlg_eval_scripts/multi-bleu.perl \
    ${TEST_TARGETS_REF0} ${TEST_TARGETS_REF1} ${TEST_TARGETS_REF2} ${TEST_TARGETS_REF3} \
    ${TEST_TARGETS_REF4} ${TEST_TARGETS_REF5} ${TEST_TARGETS_REF6} ${TEST_TARGETS_REF7} < ${TEST_PRED}

```
The score is:
'BLEU = 45.63, 75.6/52.6/38.1/28.6 (BP=1.000, ratio=1.023, hyp_len=43863, ref_len=42892)', which is not so good as in the paper (52.78 BLEU on the all test set). I wonder if there is any errors in the above instructions.

P.S I find the referece texts in the evaluating are the processed (tokenized and lower) text rather not original text. I doubt it will have a positive influence to the evaluation result slightly.

**Comment by zhaochaocs** [comment]
Hi,

To evaluate on all test set, we use non-delexicalized model for seen data and delexicalized model for the unseen data, since neural models usually generalize poorly to unseen data and delexicalization can help improve generalizability.

For the referenced text, we evaluate all the systems using the same reference texts so it should be fine.
